### PR TITLE
Actually adding historic incidents to global map.

### PR DESCRIPTION
**Pull request by captncraig**
This needs to be added to fix historic incidents after merging previous pr.

### CODE REVIEW

**Comment by captncraig** [comment]
The situation we have now is that incidents were "created" for 50k or so historic incidents, but not added to the permanent data structure. In order to fix this I propose we 

1. Stop bosun.
2. Pull the state file locally and run the following program to delete all incident data from our state file:
```
package main

import (
	"bosun.org/_third_party/github.com/boltdb/bolt"
	"fmt"
)

func main() {
	db, err := bolt.Open("bosun.state", 0600, nil)
	if err != nil {
		log.Fatal(err)
	}
	defer db.Close()
	err := db.Update(func(tx *bolt.Tx) error {
		bucket := tx.Bucket([]byte("bindata"))
		return bucket.Delete([]byte("incidents"))
	})
	fmt.Println(err)
}
```
3. Restart bosun with the new build.

**Comment by mjibson** [comment]
Another possible fix: add some code that always runs createInitialIncidents. We deploy it once, then can remove it. This would not require running a special program with the bosun state file.

File: cmd/bosun/sched/check_test.go [hunk 1]
```diff
@@ -148,6 +148,67 @@ func TestCheckSilence(t *testing.T) {
...
+	expect(1)
+
+	r.Events[ak].Status = StWarning
+	r.Events[ak].IncidentId = 0
+	s.RunHistory(r)
+	expect(1)
+	r.Events[ak].Status = StNormal
+	r.Events[ak].IncidentId = 0
+	s.RunHistory(r)
+	err = s.Action("joe", "abc", ActionClose, ak)
```
Comment (mjibson) [inline cmd/bosun/sched/check_test.go]: these strings can both be `""`

File: cmd/bosun/sched/check_test.go [hunk 2]
```diff
@@ -148,6 +148,67 @@ func TestCheckSilence(t *testing.T) {
...
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	r.Events[ak].Status = StWarning
+	r.Events[ak].IncidentId = 0
+	s.RunHistory(r)
+	expect(2)
+
```
Comment (mjibson) [inline cmd/bosun/sched/check_test.go]: remove empty line

File: cmd/bosun/sched/check_test.go [hunk 3]
```diff
@@ -148,6 +148,67 @@ func TestCheckSilence(t *testing.T) {
...
+	r.Events[ak].Status = StNormal
+	r.Events[ak].IncidentId = 0
+	s.RunHistory(r)
+	expect(1)
+
+	r.Events[ak].Status = StWarning
+	r.Events[ak].IncidentId = 0
+	s.RunHistory(r)
+	expect(1)
```
Comment (mjibson) [inline cmd/bosun/sched/check_test.go]: newline after this to be consistent with other expect calls

File: cmd/bosun/sched/check_test.go [hunk 4]
```diff
@@ -148,6 +148,67 @@ func TestCheckSilence(t *testing.T) {
...
+	r.Events[ak].Status = StNormal
+	r.Events[ak].IncidentId = 0
+	s.RunHistory(r)
+	err = s.Action("joe", "abc", ActionClose, ak)
+
```
Comment (mjibson) [inline cmd/bosun/sched/check_test.go]: no newline here since the err was assigned just on the line before

File: cmd/bosun/sched/check_test.go [hunk 5]
```diff
@@ -148,6 +148,67 @@ func TestCheckSilence(t *testing.T) {
 	}
 }
 
+func TestIncidentIdsABC(t *testing.T) {
+	s := new(Schedule)
+	c, err := conf.New("", `
+		template t {
```
Comment (mjibson) [inline cmd/bosun/sched/check_test.go]: i don't see a need for the template or notification here

File: cmd/bosun/sched/check_test.go [hunk 6]
```diff
@@ -148,6 +148,67 @@ func TestCheckSilence(t *testing.T) {
...
+			subject = 1
+		}
+		notification n {
+			print = true
+		}
+		alert a {
+			warnNotification = n
+			warn = 1
```
Comment (mjibson) [inline cmd/bosun/sched/check_test.go]: the warn can also go

**Comment by mjibson** [comment]
To explain more what I was thinking above: we always run createHistoricIncidents, and have it always reassign all incident IDs. This will get them in the right order. Yes, it'll change some, but that's fine since this is so new.

**Pull request merged**